e2e/support/helpers/e2e-notebook-helpers.ts

```diff
@@ -0,0 +1,201 @@
+import { popover } from "e2e/support/helpers/e2e-ui-elements-helpers";
+
+/**
+ * Switch to a notebook editor from a simple query view (aka "chill mode").
+ */
+export function openNotebook() {
+  return cy.findByTestId("qb-header-action-panel").icon("notebook").click();
+}
+
+/**
+ * Select a specific notebook step like filter, join, breakout, etc.
+ *
+ * @see {@link https://github.com/metabase/metabase/pull/17708#discussion_r700082403}
+ */
+export function getNotebookStep(
+  type:
+    | "summarize"
+    | "join"
+    | "expression"
+    | "filter"
+    | "sort"
+    | "limit"
+    | "data",
+  { stage = 0, index = 0 } = {},
+): Cypress.Chainable<JQuery<HTMLElement>> {
+  return cy.findByTestId(`step-${type}-${stage}-${index}`);
+}
+
+/**
+ * @summary Visualize notebook query results.
+ *
+ * This helper intelligently waits for the query to load, and gives you an option
+ * to assert on the waited query response.
+ *
+ * @example
+ * visualize();
+ *
+ * visualize(response => {
+ *   expect(response.body.error).to.not.exist;
+ * });
+ */
+export function visualize(callback?: (response?: object) => void) {
```

Comment:
Ok, so it's already useful to type the responses.

Eventually, when we start to convert `services.js` to TS, we will also need these types there. We should start adding them in the codebase and e2e tests should just import these types from there.

Reply:
After a call with @kamilmielnik , the deal was to use this
https://github.com/metabase/metabase/pull/36474/commits/b13f8dd2c2fa9db06d71a395277905db14e37265